collections/codec/time.go

```diff
@@ -0,0 +1,54 @@
+package codec
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+)
+
+var timeSize = 8
```

Comment:
idk if 8 bytes gives you the precision you're looking for, assuming the time is represented as Join(seconds,milliseconds)

Reply:
Using a 64-bit integer to directly represent a Unix timestamp with millisecond precision usually provides sufficient precision and range, eliminating the need to separately represent seconds and milliseconds.